README.md

```diff
@@ -527,6 +527,35 @@ Felizmente, há uma solução melhor para esse dilema: executar testes de contra
 
 ## <a id="sec-ui-tests"></a>UI Tests
 
+A maioria dos aplicativos tem algum tipo de interface de usuário. Normalmente, estamos falando de uma interface da Web no contexto de aplicativos da Web. Muitas vezes, as pessoas esquecem que uma API REST ou uma interface de linha de comando é tanto uma interface de usuário quanto uma interface de usuário da web sofisticada.
+
+UI tests testa se a interface do usuário do seu aplicativo funciona corretamente. A entrada do usuário deve acionar as ações corretas, os dados devem ser apresentados ao usuário, o estado da interface do usuário deve mudar conforme o esperado.
+
+![](/assets/ui_tests.png)
+
+UI Tests e end-to-end tests às vezes (como no caso de Mike Cohn) dizem ser a mesma coisa. Para mim, isso combina duas coisas que são conceitos bastante ortogonais.
+
+Sim, testar seu aplicativo de ponta a ponta geralmente significa conduzir seus testes por meio da interface do usuário. O inverso, porém, não é verdadeiro.
+
+O teste da interface do usuário não precisa ser feito de ponta a ponta. Dependendo da tecnologia que você usa, testar sua interface de usuário pode ser tão simples quanto escrever alguns testes de unidade para seu código javascript front-end com seu back-end eliminado.
```

Comment:
Colocar front-end e back-end em itálico. "Stubbed" eu acho que ficaria melhor se fosse traduzido por "substituído por um stub", por exemplo.